### PR TITLE
[Obsolete, see #147] Not using silent rules; improve runner compilation parallelism

### LINKED ISSUE #80: Makefiles: Avoid silent actions if possible
The makefiles currently have a lot of silent actions right now, which makes it harder understand what is happening and to debug what is going wrong when things fail. Often there is an `@echo` that describes what the following command does. We should let the commands speak for themself.

Let's avoid these to have a simpler and easier to use build-system. Here is an example on a subset of Makefile rules; instead of this:

```Makefile
clean:
	@echo -e "Removing $(OUT_DIR)"
	@rm -rf $(OUT_DIR)
	@echo -e "Removing $(TESTS_DIR)/generated/"
	@rm -rf $(TESTS_DIR)/generated/

report: init info tests
	@echo -e "\nGenerating report"
	@./tools/sv-report
	@echo -e "\nDONE!"

$(OUT_DIR)/logs/$(1)/$(2).log: $(TESTS_DIR)/$(2)
	@mkdir -p $(OUT_DIR)/logs/$(1)/$(dir $(2))
	@./tools/runner --runner $(1) --test $(2)
```

The following will be easier to read, debug and maintain:

```Makefile
clean:
	rm -rf $(OUT_DIR)
	rm -rf $(TESTS_DIR)/generated/

report: init info tests
	./tools/sv-report

$(OUT_DIR)/logs/$(1)/$(2).log: $(TESTS_DIR)/$(2)
	mkdir -p $(OUT_DIR)/logs/$(1)/$(dir $(2))
	./tools/runner --runner $(1) --test $(2)
```

(Possibly the only place where a silent action is useful is for `@echo` itself.)

### PR DESCRIPTION
**Pull request by hzeller**
* The silent rules made it hard to follow what is going on.
 * If we use `$(MAKE)` directly in a rule, it is able to inherit
   other configuration from the inherited make, such as -j; this
   makes compilation much faster in these cases.
  * Instead of `cp`, use `install` to install binaries
     * It prevents issues when the binary is in use at the time
          for some reason.
      * With -D, we can make sure that directories are created, so
          no need for our own `mkdir -p`
    
Fixes #80

### CODE REVIEW

**Comment by piotr-binkowski** [comment]
It looks like one commit is missing a sign-off, I will merge this after its fixed

**Comment by hzeller** [comment]
Ok, since the missing signed-off was on the first of two commits that can't be amended anymore after the fact, I created pull request #147
